ACCESS_CHALLENGE is a valid RADIUS reply.
This also fixes the Message-Authenticator calculation for such packets.
See also https://rt.cpan.org/Public/Bug/Display.html?id=158764

--- lib/Data/Radius/Packet.pm.orig	Mon Jul 15 17:27:39 2024
+++ lib/Data/Radius/Packet.pm	Tue Feb  4 11:01:36 2025
@@ -31,7 +31,7 @@ use constant {
 };
 use constant ATTR_MSG_AUTH_ZERO => pack('C C', ATTR_MSG_AUTH, ATTR_MSG_AUTH_LEN) . ("\x0" x (ATTR_MSG_AUTH_LEN - 2));
 
-my %IS_REPLY   = map { $_ => 1 } (ACCESS_ACCEPT, ACCESS_REJECT, DISCONNECT_ACCEPT, DISCONNECT_REJECT, COA_ACCEPT, COA_REJECT);
+my %IS_REPLY   = map { $_ => 1 } (ACCESS_ACCEPT, ACCESS_CHALLENGE, ACCESS_REJECT, DISCONNECT_ACCEPT, DISCONNECT_REJECT, COA_ACCEPT, COA_REJECT);
 my %IS_REQUEST = map { $_ => 1 } (ACCESS_REQUEST, ACCOUNTING_REQUEST, DISCONNECT_REQUEST, COA_REQUEST);
 
 my %IS_ACCOUNTING = map { $_ => 1 } (ACCOUNTING_REQUEST, ACCOUNTING_RESPONSE);
